### PR TITLE
docs: added firewall requirements

### PR DESCRIPTION
**Pull request by jibel**
Documented firewall requirements with a diagram and a table.
Added a link to the requirements from the how to set up page.

UDENG-2446

### CODE REVIEW

File: docs/reference/firewall_requirements.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+![Firewall considerations.](./assets/firewall_requirements.png)
+
+The following table lists the default ports and protocols used by Ubuntu Pro for WSL:
+
+| Description | Client System | Server System | Protocol | Default Port | Target address |
+|-------------|---------------|---------------|----------|--------------|----------------|
+| Required for online installation of WSL instances.|Windows Host / Pro Agent|MS Store | tcp | 443 (https) | See [Microsoft documentation](https://learn.microsoft.com/en-us/microsoft-store/prerequisites-microsoft-store-for-business) for a list of addresses to allow. |
+| Ubuntu Pro enablement |Windows Host / Pro Agent |Canonical Contract Server |tcp |443 (https) | contracts.canonical.com |
+| Landscape management | Windows Host / Pro Agent | Landscape Server | tcp | 6554 (grpc) | On-premise Landscape address |
```
Comment (CarlosNihelton) [inline docs/reference/firewall_requirements.md:15]:
```suggestion
| Landscape management | Windows Host / Pro Agent | Landscape Server | tcp | 6554 (gRPC) | On-premise Landscape address |
```

File: docs/reference/firewall_requirements.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+| WSL instance management on the Windows host. Firewall rules set up at installation time of the WSL Pro agent. | WSL Instance / wsl-pro-service | Windows Host / Pro Agent | tcp | 49152-65535 (dynamic) | Hyper-V Virtual Ethernet Adapter IP |
+| Ubuntu Pro. For air-gapped installation refer to the [Ubuntu Pro documentation](https://canonical-ubuntu-pro-client.readthedocs-hosted.com/en/latest/explanations/using_pro_offline/). | WSL Instance / Ubuntu Pro client | Canonical Contract Server | tcp | 443 (https) | contracts.canonical.com |
+| Landscape |  WSL Instance / Ubuntu Pro client | Landscape Server | tcp | 443 (https) | On-premise Landscape address |
+
+> Access to the contract server and Landscape server is required for proper operation of Ubuntu Pro for WSL. If the client system is behind a proxy, ensure that the proxy is configured to allow the required connections.
+
+> Access to the Microsoft Store is required for the online installation of WSL instances. Without it Ubuntu Pro for WSL will still be functional but it will not be possible to install WSL instances centrally from Landscape. In this case WSL instances have to be installed manually on the Windows hosts.
```
Comment (CarlosNihelton) [inline docs/reference/firewall_requirements.md:22]: I think those notes would work better as footnotes cross-referenced from the specific items in the table that relate to them.

File: docs/custom_conf.py [hunk 3]
```diff
@@ -123,7 +123,10 @@
     'http://127.0.0.1:8000',
 
     # Linkcheck does not have access to the repo
-    'https://github.com/canonical/ubuntu-pro-for-wsl/*'
+    'https://github.com/canonical/ubuntu-pro-for-wsl/*',
+
+    # Not a URL in the reference document
+    r'http://contracts.canonical.com'
```
[the remark this answers is not in the document]
Comment (jibel) [reply docs/custom_conf.py:129]: It's really http, the link checker identifies 'contracts.canonical.com' in the doc as a URL and extends it to 'http://contracts.canonical.com', which obviously doesn't exist.

File: docs/reference/firewall_requirements.md (hunk 1 above)
Comment (jibel) [reply docs/reference/firewall_requirements.md:15]: I used small caps for all the protocols in the table.

**Review by CarlosNihelton — APPROVED** [review]
The broken link is unrelated, and after a second attempt the link check will certainly succeed.

**Pull request merged**